### DISCUSSION TITLE
Why stop Text encoder training with external captions when finetuning UNet ?

### DISCUSSION THREAD

**Discussion by MiJiw** [TheLastBen/fast-stable-diffusion#1322] [Q&A]
Hi TheLastBen,
Thank you for your great work !

I have a question about the stop_text_encoder_training when using external captions.
In your custom train_dreambooth.py, you have these lines that shut off the use of captions after :
>if args.external_captions and global_step == args.stop_text_encoder_training and global_step >= 5:
      subprocess.call('mv '+args.captions_dir+' '+args.captions_dir+'off', shell=True)

Could you elaborate on the logic behind this choice ?

I commented them out for now as it seems to give me better models but I might be missing something (I usually train concept text encoder and UNet in the same run). I work with datasets of 100+ images with a common subject but with very different appearances and environments, thus it seems to help to generate specific combination when prompting.

**Comment by TheLastBen** [comment]
when using captions, the text encoder is not finetuned, so the number of steps set in the text_encoder_steps means the number of steps for which to pass the captions to the unet, if you set it to 0, it will use the captions for the whole training.
